### PR TITLE
Enable React Strict Mode by default in tests

### PR DESCRIPTION
**Pull request by lazytype**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Investigate enabling React strict mode in tests](https://jira.mongodb.org/browse/PD-736)

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

**Comment by lazytype** [comment]
Some tests are failing now (so it looks like it's working), but need to fix them before merging.

File: scripts/jestSetup.js [hunk 1]
```diff
@@ -15,3 +15,13 @@ global.IntersectionObserver = class {
   unobserve() {}
   disconnect() {}
 };
+
+const MockReact = jest.requireActual('react');
+const MockReactDOM = jest.requireActual('react-dom');
```
Comment (hswolff) [inline scripts/jestSetup.js:20]: nit - aren't these the actual modules? The variable names are wrong?

Comment (lazytype) [reply scripts/jestSetup.js:20]: jest will throw an error if you try to access variables outside the mock factory's scope unless you whitelist them by prefixing them with "mock". 🤷‍♀️
https://github.com/facebook/jest/blob/5fe8b0/packages/babel-plugin-jest-hoist/src/index.ts#L140-L152

Comment (hswolff) [reply scripts/jestSetup.js:20]: lovely

**Pull request merged**